store/serializers.py

```diff
@@ -611,17 +613,24 @@ def create(self, validated_data):
 
             if membership_orderlines:
                 need_transaction = True
-                today = timezone.now().date()
+                # Allow to buy membership, 1 month before end of membership
+                today = timezone.now().date() + relativedelta(months=1)
                 if user.membership and user.membership_end > today:
                     raise serializers.ValidationError({
                         'non_field_errors': [_(
                             "You already have an active membership."
                         )]
                     })
                 user.membership = membership_orderlines[0].content_object
-                user.membership_end = (
-                        timezone.now().date() + user.membership.duration
-                )
+                # If the user has already a membership end,
+                # we add the new membership duration to it
+                if user.membership_end:
+                    user.membership_end = \
+                        user.membership_end + user.membership.duration
```

Comment:
This logic does not work if a User had a membership a long time ago and want to buy a new one after 1 year without membership.

We need to check if `membership_end > today` and in this case only we add one year to this membership_end

PS: Please add tests to cover this forgotten specification